napari/components/viewer_model.py

```diff
@@ -384,6 +384,7 @@ def _on_layers_change(self, event):
             self.dims.ndim = ndim
             for i in range(ndim):
                 self.dims.set_range(i, (world[0, i], world[1, i], ss[i]))
+        self.cursor.position = (0,) * self.dims.ndim
```

Comment:
Why is this change necessary?

Reply:
it's something that we should have been doing anyway - not critical for this PR to work, but otherwise when adding layers the cursor position could get the wrong dimensionality and be off.